### PR TITLE
Explain the lack of impact on workers

### LINKED ISSUE #4: Shared/service workers
It's worth pointing out why the agent clusters created for these already meet the requirements. In effect this is really about window agent isolation.

### PR DESCRIPTION
**Pull request by domenic**
Closes #4.

@annevk, would you mind reviewing?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,15 +140,27 @@ In particular, when creating a document or worker, we use the following in place
...
 
 You can see a more full analysis of what results this algorithm produces in our [scenarios document](./scenarios.md).
 
 As far as the web-developer–observable consequences of using origins for agent cluster keys, this will automatically cause `SharedArrayBuffer`s to no longer be shareable cross-origin, because of the agent cluster check in [StructuredDeserialize](https://html.spec.whatwg.org/multipage/structured-data.html#structureddeserialize). We'd also need to add a check to the [`document.domain` setter](https://html.spec.whatwg.org/multipage/origin.html#dom-document-domain) to make it throw. (Ideally we'd find some way of restructuring the existing checks in `document.domain` so that they are also tied to agent cluster boundaries.)
 
+### More detail on workers
+
+We mentioned above that origin isolation has no effect on workers. Here we explain why exactly that is.
+
+First note that shared and service workers are already isolated into their own agent clusters. The only other things that exist in those agent clusters are potential nested dedicated workers. Everything is already same-origin within those shared and service worker agent clusters.
```
Comment (annevk) [inline README.md]: Not quite everything, but everything that creates an agent, yes?

File: README.md [hunk 2]
```diff
@@ -137,15 +140,27 @@ In particular, when creating a document or worker, we use the following in place
...
+
+What remains to consider is dedicated workers which are spawned directly by documents. Those end up in the agent cluster of their owner document, and the allocation of that agent cluster is impacted by origin isolation, which makes it origin-keyed instead of site-keyed.
+
+However, being origin-keyed has no observable impact for code inside a dedicated worker. All access to other realms is asynchronous inside a dedicated worker, so `document.domain` does not apply. And there's no way for dedicated worker code to send a `SharedArrayBuffer` to a same-site cross-origin destination, because the only thing it can communicate with is its parent document, or further nested dedicated workers, all of which are same-origin. (Note that even `BroadcastChannel`, which bypasses the need to have a direct reference to the destination, is restricted to same-origin communications.)
```
Comment (annevk) [inline README.md]: With cooperation of the parent document you can use `MessageChannel` to communicate same-site cross-origin. However, due to the keying of the parent document's agent cluster that would be a different agent cluster and therefore fail for `SharedArrayBuffer`.

File: README.md (hunk 1 above)
Comment (domenic) [reply README.md]: I guess I meant "all other agents are already same-origin", so this should probably be merged with the previous sentence. Will update.

File: README.md (hunk 2 above)
Comment (domenic) [reply README.md]: Right, that's true. I explained that scenario a bit; let me know what you think of the result.

**Review by annevk — APPROVED** [review]

**Pull request merged**